Possible bug: tests/test_0652_dask-for-awkward.py in uproot tests fails on dask_awkward main

Using dask_awkward main branch:
```python3
test_path = (
    skhep_testdata.data_path("uproot-sample-6.08.04-uncompressed.root") + ":sample"
)
ttree = uproot.open(test_path)
    
ak_array = ttree.arrays()
dak_array = uproot.dask(test_path, library="ak")

assert len(ak_array) == len(dak_array)
```
results in:
```
AssertionError: assert 30 == 0
+  where 30 = len(<Array [{n: 0, b: True, ab: [...], ...}, ...] type='30 * {n: int32, b: bool...'>)
+  and   0 = len(dask.awkward<from-uproot, npartitions=1>)
```

opening this here despite it possibly being in a deeper library @agoose77 @jpivarski

--- a/src/dask_awkward/layers/layers.py
+++ b/src/dask_awkward/layers/layers.py
@@ -136,7 +136,6 @@ class AwkwardInputLayer(Blockwise):
 
     def project_columns(self, columns: list[str]) -> AwkwardInputLayer:
         if hasattr(self.io_func, "project_columns"):
-
             # TODO: make project_columns call sites never pass in an
             # empty list.
             if len(columns) == 0:

--- a/src/dask_awkward/layers/layers.py
+++ b/src/dask_awkward/layers/layers.py
@@ -136,6 +136,12 @@ class AwkwardInputLayer(Blockwise):
 
     def project_columns(self, columns: list[str]) -> AwkwardInputLayer:
         if hasattr(self.io_func, "project_columns"):
+
+            # TODO: make project_columns call sites never pass in an
+            # empty list.
+            if len(columns) == 0:
+                columns = self._meta.fields[:1]
+
             io_func = self.io_func.project_columns(columns)
             return AwkwardInputLayer(
                 name=self.name,

